packages/mui-material/src/Select/Select.d.ts

```diff
@@ -148,9 +148,45 @@ export interface SelectProps<Value = unknown>
    * The variant to use.
    * @default 'outlined'
    */
-  variant?: 'standard' | 'outlined' | 'filled';
+  variant?: 'filled' | 'standard' | 'outlined';
 }
 
+export interface FilledSelectProps extends Omit<FilledInputProps, 'value' | 'onChange'> {
+  /**
+   * The variant to use.
+   * @default 'outlined'
+   */
+  variant: 'filled';
+}
+
+export interface StandardSelectProps extends Omit<InputProps, 'value' | 'onChange'> {
+  /**
+   * The variant to use.
+   * @default 'outlined'
+   */
+  variant: 'standard';
+}
+
+export interface OutlinedSelectProps extends Omit<OutlinedInputProps, 'value' | 'onChange'> {
+  /**
+   * The variant to use.
+   * @default 'outlined'
+   */
+  variant: 'outlined';
+}
+
+export type SelectVariants = 'outlined' | 'standard' | 'filled';
```

Comment:
How about importing `TextFieldVariants` from [here](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/TextField/TextField.d.ts#L229) instead?

Reply:
That sounds reasonable 👍 I am unavailable to make the change at the moment since I am on vacation however I will implement it next week.

Reply:
@mj12albert I think it's important to define them separately, even if the type values are the same. `TextField` and `Select` component are unrelated.

Reply:
Ok I will keep it as it is at the moment then.